### ISSUE TITLE
Bow gets used instead of Blanket when Sleeping

### ISSUE THREAD

**Issue by ArikVhedrovix** [CleverRaven/Cataclysm-DDA#19244]
For some strange reason my character seems to prefer to snuggle his bow instead of the blanket he is carrying. Let's not judge too harshly, cataclysms do strange things to people... 
![screen shot 2016-11-11 at 1 49 19 pm](https://cloud.githubusercontent.com/assets/5952322/20228612/a4b775d6-a817-11e6-90bc-7179c0061315.png)

**Comment by Shoes01** [comment]
Are you wearing the blanket? Looking at `player::is_snuggling()`, it appears that you "aren't" snuggling your blanket at all, which is worrying. 

Also, despite there being an `is_armor` check, perhaps something more strict should be done to disallow a bow from being used as a snuggle buddy.

**Comment by ArikVhedrovix** [comment]
After doing more tests I am even more confused. I was going to test this bug under various conditions such as on my cot and rollmat in addition to in the car seat where I had observed the bug first. I tried sleeping in the seat with the bow stowed in the seat and the bow was used to sleep. Then I put up a cot and rollmat to test. I slept on the cot first with bow on ground, got same bugged result. Then before I could sleep on the rollmat I got a "Torso getting warm" warning and since that I can't get my character to use the bow, with or without torso warmth effect. I was planning to test if holding the bow in inventory would give the same results as having it on the ground, but the torso warmth warning seems to have made my character stop using the bow.

Here is a view of all the stuff on the car at the seat location:
![car view](https://cloud.githubusercontent.com/assets/5952322/20248281/1faf15b6-a9a6-11e6-9444-cb177704efb9.png)

Here is my inventory:
![inv](https://cloud.githubusercontent.com/assets/5952322/20248283/283b416e-a9a6-11e6-8c03-57408f6cfdda.png)

Here are the message logs:
Here you can see me sleeping in the car with bow
![one](https://cloud.githubusercontent.com/assets/5952322/20248284/3b5aec04-a9a6-11e6-80e0-b3829a0212b3.png)

Now I set up cot and rollmat and slept on cot giving bug...
![two](https://cloud.githubusercontent.com/assets/5952322/20248293/469731fe-a9a6-11e6-93bb-410ac434790f.png)

Now I got Torso warm warning and now no longer sleep with bow anywhere....
![three](https://cloud.githubusercontent.com/assets/5952322/20248295/5a489bb6-a9a6-11e6-8101-6336ddfd40c0.png)

So in conclusion, I am not sure why the bow began being used but apparently changing my body temp effects made me stop using it.
...

**Comment by Shoes01** [comment]
So without digging into the code, I am pretty sure you don't seek floor items for warmth if you don't need them. If you're warm and sleeping on a pile of blankets, you don't "use the piles of clothes for warmth". 

But now I'm confused as to what the bug is. In your initial message, it seemed like you were using only a bow for warmth when you had a bow and blankets on the floor. Was that the bug? Otherwise, my PR will fix the bug about using weapons, or other 0 warmth items, for warmth.

Let me know!

**Comment by Shoes01** [comment]
I don't think inventory items are checked for warmth, only worn and floor items. I suppose that's a bit misleading, do you think?
